### PR TITLE
pw encryption; and login built

### PR DESCRIPTION
**Pull request by KryoKorpz**
haven't implement jwt yet.

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -2,6 +2,20 @@ const User = require('../models/userModels');
 const bcrypt = require('bcrypt');
 
 const createUser = (req, res) => {
+  const { username } = req.body;
+  const passwordHash = req.password
+  const newUser = new User({ username, password: passwordHash })
+  console.log(newUser)
```
Comment (ryanhca) [inline api/controllers/user.js]: You can remove this :)

Comment (ryanhca) [reply api/controllers/user.js]: The console.log that is.

File: api/utils/middlewares.js [hunk 2]
```diff
@@ -21,15 +22,47 @@ const authenticate = (req, res, next) => {
...
+    }
+    const pwHash = user.password
+    console.log(password)
+    console.log(pwHash)
+    console.log(user)
+    bcrypt
+      .compare (password, pwHash)
+      .then((response) => {
+        if (!response) throw new Error ();
+        req.user = user;
```
Comment (ryanhca) [inline api/utils/middlewares.js]: you'll want to call next() here to toss you over into the Login route controller. Once you get through that, it aught to send your token back tot he client.

File: api/controllers/user.js (hunk 1 above)
Comment (KryoKorpz) [reply api/controllers/user.js]: oh ahah yeah; I was de-lousing :D

File: api/utils/middlewares.js (hunk 2 above)
Comment (KryoKorpz) [reply api/utils/middlewares.js]: okay awesome; yeah I hadn't completed the login yet. Thank you!!

**Comment by KryoKorpz** [comment]
Reopening a PR is hard

**Pull request closed without merging**